probe-rs/src/semihosting.rs

```diff
@@ -199,6 +212,39 @@ pub fn decode_semihosting_syscall(
                 block_address,
             )?))
         }
+
+        (SYS_WRITEC, pointer) => {
+            // Parameter points to a byte in memory.
+            let mut buf = 0;
+            core.read_8(pointer as u64, std::slice::from_mut(&mut buf))?;
+
+            let c = buf as u8 as char;
+            SemihostingCommand::WriteConsole(String::from(c))
+            // no response is given
+        }
+
+        (SYS_WRITE0, pointer) => {
+            // no response is given
+            // Parameter points to a byte in memory.
+            let mut buf = [0; 128];
```

Comment:
I guess 128 is just an arbitrary value to read from the device memory at a time? 128 seems reasonable enough to me.

Reply:
Yes, it's maybe a bit too much but the target is sitting on a breakpoint anyway so maybe we can take our time.